app/assets/javascripts/components/admin/roster_page.es6.jsx

```diff
@@ -16,20 +32,42 @@ class RosterPage extends React.Component {
   }
 
   renderPeople() {
-    return this.props.people.map((person) => this.renderPerson(person));
+    return this.state.people.map((person) => this.renderPerson(person));
   }
 
-  render () {
-    return (
-      <div className="page-wrapper">
-        <AdminHeader />
-        <div className="content-wrapper roster-page">
-          <h1 className="roster-title">Roster</h1>
-          <div className="roster-container">
-            {this.renderPeople()}
+  searchOnChange(input) {
+    if (input === "") {
+      this.loadRoster();
+    } else {
+      var route = ApiConstants.searchables.users(input);
+      var resolve = (response) => this.setState({people: response.searchables});
+      var reject = (response) => console.log(response);
+      Requester.get(
+        route,
+        resolve,
+        reject,
+      );
+    }
+  }
+
+  render() {
+    if (this.state.people == null) {
+      return (
+        <div />
+      );
+    } else {
+      return (
+        <div className="page-wrapper">
+          <AdminHeader />
+          <SearchBar onChange = {(input) => this.searchOnChange(input)} />
```

Comment:
I'm not really sure if you need to create a separate component for the search bar. I think you can just use a FormControl (check student_form file) and keep track of the input as a state in this component